prospero-common/src/main/java/org/wildfly/prospero/actions/InstallFolderUtils.java

```diff
@@ -40,6 +41,25 @@ static void verifyIsEmptyDir(Path directory) {
         }
     }
 
+    static Path toRealPath(Path symlink) {
+        Path path = symlink;
+
+        while (path != null && !(Files.exists(path) && Files.isSymbolicLink(path))) {
+            path = path.getParent();
+        }
+
+        if (path != null) {
+            final Path relativized = path.relativize(symlink);
+            try {
+                return path.toRealPath().resolve(relativized);
+            } catch (IOException e) {
```

Comment:
Not sure, but should there be another check if the path is really a symlink, before doing the `path.relativize()` and the rest?

It seems like we call this method always to sanitize the input.

T double check - IIUC the issue only is if the last segment of the path is a symlink. A symlink in the middle of the path does not cause an issue..?

Reply:
You can disregard this comment :).